### DISCUSSION TITLE
Marking morphs as invalid

### DISCUSSION THREAD

**Discussion by mortii** [mortii/anki-morphs#223] [Ideas]
>Sometimes the morphs don't make sense in the context, but sometimes they just aren't actually real "words" - it splits up two words and combines the last character of one and the first character of the next into something that is not a word.  Especially since the sentence will still be in the text that I'm using for my study plan, I'd rather have a way to mark the morph as "invalid" or "ignore" rather than suspending or deleting individual cards. Right now I'm seeing what happens if I mark them as "Names" and selecting "Ignore names found in names.txt" but this doesn't seem like a satisfactory solution.

_Originally posted by @xofm31 in https://github.com/mortii/anki-morphs/discussions/221#discussioncomment-9127055_

**Comment by mortii** [comment]
I think the easiest way to achieve this would be to duplicate the "mark as name" feature. We could have a context menu option "mark morph as invalid" and a file called "invalid-morphs.txt" or something.

**Reply by buster-blue** [reply]
I've been encountering quite a few "broken" morphs recently. a solution like this would be fine for me. I just want a way to get Ankimorphs to ignore these invalid morphs without having to put them in the names file, since they're not names so it's a confusing.

currently I'm marking them as "known" morphs, but this counts them as morphs which I don't really like.

**Comment by xofm31** [comment]
Now that I've thought about this some, and seen a lot of wrong parses from jieba, I realize what I would really like to do is to be able to fix the morphs on a card. I think your plan is to make the `all-morphs` field list all of the morphs for display only. What I would really like is to be able to delete or edit the morphs that are not correct, and then once that field is populated, Ankimorphs would read that one and not re-run the jieba. Right now I can manually change the `focus word` on the card, but that information doesn't get used when determining the known morphs.

I understand if this is too much work for the benefit, especially if the morphs produced for other languages are better.

**Reply by mortii** [reply]
Having multiple potential inputs per card would indeed cause too much complexity :(

**Reply by xofm31** [reply]
I figured, but I thought I'd mention it just in case. :)

**Comment by mortii** [comment]
[Not adding any new features](https://github.com/mortii/anki-morphs/releases/tag/v3.3.0), apologies.
